contracts/src/v0.1/RequestResponseCoordinator.sol

```diff
@@ -545,19 +580,75 @@ contract RequestResponseCoordinator is
     function pay(
         RequestCommitment memory rc,
         bool isDirectPayment,
-        uint256 startGas
+        uint256 startGas,
+        address[] memory submiters
     ) internal returns (uint256 payment) {
         if (isDirectPayment) {
-            payment = sPrepayment.chargeFeeTemporary(rc.accId, msg.sender);
+            (uint256 totalAmount, uint256 operatorAmount) = sPrepayment.chargeFeeTemporary(
+                rc.accId
+            );
+
+            uint256 paymentNoFee = calculatePaymentNoFee(
+                startGas,
+                sConfig.gasAfterPaymentCalculation
+            );
+            if (operatorAmount > 0) {
```

Comment:
I think we can simplify this a lot with certain assumptions and it will be the same code for either single or multiple oracle submissions. Also, I think it will be easier for us to assume that `operatorAmount` (in my code `operatorFee`) is price meant to be for delivering a single submission so we do not have to divide anything.

```Solidity
uint8 submitersLength = submiters.length;
for (uint8 i; i < submitersLength-1; ++i) {
    sPrepayment.payOperatorFeeTemporary(
        operatorFee,
        submiters[i]
    );
}

sPrepayment.payOperatorFeeTemporary(
    operatorFee + paymentNoFee,
    submiters[submittersLength-1]
);
```

The code above can replace everything in lines 595 to 614.